### ISSUE TITLE
Sketchy useage of `unsafe_wrap`/`unsafe_string`

### ISSUE THREAD

**Issue by oscardssmith** [wildart/LMDB.jl#41]
This library makes heavy use of  `unsafe_wrap` and `unsafe_string`, (see https://github.com/wildart/LMDB.jl/blob/bdc2df64debb30ffc4f43ae0c92107d35de0c8af/src/common.jl#L13) but doesn't seem to ensure that they aren't used on data that comes from Julia. The intention of these functions is that they are used on foreign memory only (since otherwise, you will get a double free when the original and wrapped version both get freed).

**Comment by vtjnash** [comment]
The issue isn't that (per se) but rather that the call to pointer in the constructor likely destroys the GC information, leading to use after free bugs later

**Comment by vtjnash** [comment]
The object's memory still isn't GC-rooted, due to the illegal call to `pointer` here https://github.com/JuliaDatabases/LMDB.jl/blob/a1e9607b76b6786907c1cfa20eefc4b1d044fa3d/src/common.jl#L5

**Comment by maleadt** [comment]
Yeah I'm moving too fast here. I had this fixed in a local branch, and wanted to split things up into logical PRs, but failed to do so properly. Folding the better fix into https://github.com/JuliaDatabases/LMDB.jl/pull/52
